src/digital_input_stamper.rs

```diff
@@ -0,0 +1,146 @@
+///! Digital Input 0 (DI0) reference clock timestamper
+///!
+///! This module provides a means of timestamping the rising edges of an external reference clock on
+///! the DI0 with a timer value from TIM5.
+///!
+///! This module only supports input clocks on DI0 and may or may not utilize DMA to collect
+///! timestamps.
+///!
+///! # Design
+///! An input capture channel is configured on DI0 and fed into TIM5's capture channel 4. TIM5 is
+///! then run in a free-running mode with a configured frequency and period. Whenever an edge on DI0
```

Comment:
Frequency is inverse period by definition. What do you want to say here?

Reply:
This is referring to the period register (ARR) and the prescaler (tick frequency), I'll think of better wording